impeller/entity/entity_pass.cc

```diff
@@ -485,6 +487,45 @@ bool EntityPass::Render(ContentContext& renderer,
       clip_coverage_stack);                      // clip_coverage_stack
 }
 
+// When a subpass size is close to, but still smaller than the root pass
+// size and smaller than the bounds hint, we may scale it up to the root
+// pass size. This will improve performance by improving the efficiency of
+// the render target cache, as only textures with exactly the same sizes +
+// descriptors can be recycled.
+static ISize MaybeRoundUpTextureSize(ISize subpass_size,
+                                     ISize root_pass_size,
+                                     std::optional<Rect> bounds_limit,
+                                     const Matrix& subpass_transform) {
+  // If the subpass is already bigger than the root pass size,
+  // return the existing subpass size.
+  if (subpass_size.width > root_pass_size.width ||
+      subpass_size.height > root_pass_size.height) {
+    return subpass_size;
+  }
+
+  // If there is a bounds limit and it is tigher than the root pass size,
+  // return the existing subpass size. This case could be removed if we
+  // conditionally inserted clips/scissor instead.
+  if (bounds_limit.has_value()) {
+    auto user_bounds_size =
+        bounds_limit->TransformBounds(subpass_transform).GetSize();
```

Comment:
Would you mind adding a method like `std::optional<Rect> EntityPass::GetTransformedBoundsLimit()`? `GetSubpassCoverage` can use it directly and we can pass in `transformed_bounds_limit` for this static method. Just to make it easier to not screw this up later on during refactors.

Reply:
I would have recommended we just apply the transform when it's set, but unfortunately the transform might end up getting modified if this `EntityPass` is appended to another `EntityPass` via DrawPicture...

Reply:
I think we're need to remove drawPicture support soon, it doesn't actually work and was added for an aiks canvas API that didn't work out.